File: java/src/main/java/org/example/matrix/Matrix.java
package org.example.matrix;

import java.util.Random;

public class Matrix {
    public static double[][] randomMatrix(int n, long seed) {
        Random rng = new Random(seed);
        double[][] M = new double[n][n];
        for (int i=0;i<n;i++)
            for (int j=0;j<n;j++)
                M[i][j] = rng.nextDouble();
        return M;
    }

    public static double[][] matmul_ijk(double[][] A, double[][] B) {
        int n = A.length;
        double[][] C = new double[n][n];
        for (int i=0;i<n;i++) {
            for (int k=0;k<n;k++) {
                double aik = A[i][k];
                for (int j=0;j<n;j++) {
                    C[i][j] += aik * B[k][j];
                }
            }
        }
        return C;
    }
}

File: python/matrix.py
from typing import List

def matmul_ijk(A: List[List[float]], B: List[List[float]]) -> List[List[float]]:
    n = len(A)
    C = [[0.0]*n for _ in range(n)]
    for i in range(n):
        Ai = A[i]
        Ci = C[i]
        for k in range(n):
            aik = Ai[k]
            Bk = B[k]
            for j in range(n):
                Ci[j] += aik * Bk[j]
    return C